akka-projection-alpakka-kafka/src/main/scala/akka/projection/scaladsl/kafka/KafkaProjection.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2019-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.scaladsl.kafka
+
+import akka.kafka.CommitterSettings
+import akka.kafka.ConsumerMessage.{ CommittableMessage, CommittableOffset }
+import akka.kafka.scaladsl.Consumer.{ Control, DrainingControl }
+import akka.kafka.scaladsl.{ Committer, Consumer }
+import akka.projection.scaladsl._
+import akka.stream.Materializer
+import akka.stream.scaladsl.{ Keep, Sink, Source }
+import akka.{ Done, NotUsed }
+import org.apache.kafka.clients.consumer.ConsumerRecord
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+object KafkaProjection {
+
+  def apply[K, V](
+      src: Source[ConsumerRecord[K, V], Control],
+      handler: EventHandler[V, Future[Done]]): Projection[ConsumerRecord[K, V], V, NotUsed, Future[Done]] =
+    new ConsumerRecordProjection(src, handler)
+
+  def apply[K, V, R](
+      sourceProvider: SourceProvider[Long, ConsumerRecord[K, V], Control],
+      handler: EventHandler[V, R],
+      projectionRunner: ProjectionRunner[Long, R]): Projection[ConsumerRecord[K, V], V, Long, R] = {
+    new ConsumerRecordProjectionWithRunner(sourceProvider, handler, projectionRunner)
+  }
+
+  def atLeastOnce[K, V](
+      committableSource: Source[CommittableMessage[K, V], Control],
+      committerSettings: CommitterSettings,
+      handler: EventHandler[V, Future[Done]])
+      : Projection[CommittableMessage[K, V], V, CommittableOffset, Future[Done]] = {
+    new AtLeastOnceKafkaProjection(committableSource, committerSettings, handler)
+  }
```

Comment:
This is one uses the CommiterFlow. Accepts any source emitting `CommittableMessage`. 

Note that the event handler is typed on `V`, the value. We may consider a variant the receives the Key and the Value.